### PR TITLE
Rotate Drivetrain Command

### PR DESCRIPTION
**Pull request by OhadLevine**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/drivetrain/RotateDrivetrain.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    drivetrain.arcadeDrive(pidController.trigonCalculate(drivetrain.getAngle(), desiredAngle.getAsDouble()), 0);
+  }
+
+  @Override
+  public void end(boolean interrupted) {
+    drivetrain.stopMove();
+  }
+
+  @Override
+  public boolean isFinished() {
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/drivetrain/RotateDrivetrain.java]: end after isFinished

File: src/main/java/frc/robot/subsystems/drivetrain/RotateDrivetrain.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import java.util.function.DoubleSupplier;
+import frc.robot.utils.TrigonPIDController;
+
+import static frc.robot.Robot.drivetrain;
+import static frc.robot.Robot.robotConstants;
+
+public class RotateDrivetrain extends CommandBase {
+  TrigonPIDController pidController;
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/drivetrain/RotateDrivetrain.java]: private

File: src/main/java/frc/robot/subsystems/drivetrain/RotateDrivetrain.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    addRequirements(drivetrain);
+    pidController = new TrigonPIDController(robotConstants.controlConstants.drivetrainRotateSettings);
+    pidController.enableContinuousInput(-180, 180);
+  }
+
+  /**
+   * Rotates the drivetrain in place. This constructor is used for tuning the PID
+   * in the smartdashboard.
+   */
+  public RotateDrivetrain() {
```
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/drivetrain/RotateDrivetrain.java]: add to the data container this command so it will be ready to config the PID

**Review by HilelAmiel — CHANGES_REQUESTED** [review]
few things

File: src/main/java/frc/robot/subsystems/drivetrain/RotateDrivetrain.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  }
+
+  @Override
+  public void initialize() {
+    pidController.reset();
+  }
+
+  @Override
+  public void execute() {
+    drivetrain.arcadeDrive(pidController.trigonCalculate(drivetrain.getAngle(), desiredAngle.getAsDouble()), 0);
```
Comment (EliyaFishman) [inline src/main/java/frc/robot/subsystems/drivetrain/RotateDrivetrain.java]: you can replace `drivetrain.arcadeDrive` with `drivetrain.move`

**Review by EliyaFishman — CHANGES_REQUESTED** [review]

**Comment by EliyaFishman** [comment]
[Drivetrain Command: turn in place](https://trello.com/c/TubvyR1N/111-drivetrain-command-turn-in-place)

**Review by HilelAmiel — APPROVED** [review]

**Review by EliyaFishman — APPROVED** [review]

**Pull request merged**